check-lint

```diff
@@ -1,9 +1,73 @@
 #!/bin/sh -e
 
 TOOL_CMD=pylint
-PATH=$(basename $0)/utils:$PATH
-CONFIG_PATH="$(dirname $0)/default_configs/pylintrc"
-FILES=$(git ls-files '*.py')
+BASE_DIR=$(dirname "$0")
+CONFIG_FILE="$BASE_DIR/../avocado-static-checks.conf"
 
-echo "** Running $TOOL_CMD (using config from $CONFIG_PATH)..."
-python3 -m pylint --rcfile=$CONFIG_PATH $FILES
+# Default config path
+DEFAULT_CONFIG_PATH="$BASE_DIR/default_configs/pylintrc"
+
+# Collect all Python files in the repository
+ALL_FILES=$(git ls-files '*.py')
+
+# Create a temporary file to store files checked with custom configs
+CUSTOM_CONFIG_FILES=$(mktemp)
+
+# Initialize a flag to track if we're within the [lint] section
+in_lint_section=false
+
+# Check if the avocado-static-checks.conf file exists
+if [ -f "$CONFIG_FILE" ]; then
+    # Read and process each line in the config file
+    while IFS=':' read -r directory config_path; do
+        # Handle section headers
+        if [[ "$directory" =~ ^\s*\[lint\]\s*$ ]]; then
+            in_lint_section=true
+            continue
+        elif [[ "$directory" =~ ^\s*\[.*\]\s*$ ]]; then
+            in_lint_section=false
+            continue
+        fi
+
+        # Process lines only within the [lint] section
+        if $in_lint_section; then
+            # Trim whitespace
+            CONFIG_PATH=$(echo "$config_path" | xargs)
+            DIRECTORY_PATH=$(echo "$directory" | xargs)
+
+            # Check if the directory exists
+            if [ ! -d "$BASE_DIR/../$DIRECTORY_PATH" ]; then
+                echo "Error: Directory '$DIRECTORY_PATH' specified in the config file does not exist."
+                exit 1
+            fi
+
+            # Check if the config path exists
+            if [ ! -f "$BASE_DIR/../$CONFIG_PATH" ]; then
+                echo "Error: Config file '$CONFIG_PATH' specified for directory '$DIRECTORY_PATH' does not exist."
+                exit 1
+            fi
+
+            # Get all Python files in the specified directory
+            FILES=$(git ls-files "$DIRECTORY_PATH/*.py")
+
+            if [ -n "$FILES" ]; then
+                echo "** Running $TOOL_CMD on directory '$DIRECTORY_PATH' with config from '$CONFIG_PATH'..."
+                python3 -m pylint --rcfile="$BASE_DIR/../$CONFIG_PATH" $FILES
+
+                # Add the files to the custom config list
+                echo "$FILES" >> "$CUSTOM_CONFIG_FILES"
+            fi
+        fi
+    done < "$CONFIG_FILE"
+fi
+
+# Calculate remaining files that weren't checked with a custom config
+REMAINING_FILES=$(comm -23 <(echo "$ALL_FILES" | sort) <(sort "$CUSTOM_CONFIG_FILES"))
+
+if [ -n "$REMAINING_FILES" ]; then
+    echo "** Running $TOOL_CMD with default config on remaining files..."
+    python3 -m pylint --rcfile="$DEFAULT_CONFIG_PATH" $REMAINING_FILES
+fi
+
+# Clean up the temporary file
+rm "$CUSTOM_CONFIG_FILES"
```

Comment:
Hi @harvey0100, actually your change hasn't fixed this problem, because you still won't do the clean-up, when some of the paths doesn't exist and your script will finish with `exit 1`. IMO, the best way how to solve this is to don't use the tempfile for storing runtime data and use array instead.